### PR TITLE
Change some usage of RawKeyEvent to KeyEvent in preparation for deprecation

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This converts some usages of `RawKeyEvent` to `KeyEvent` to prepare the repo for deprecation of `RawKeyEvent`, and swaps out the `raw_keyboard.dart` manual test for `hardware_keyboard.dart`.

## Related Issues
 - https://github.com/flutter/flutter/issues/136419

## Tests
 - Just refactoring code, no semantic changes.

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: refactor

File: dev/manual_tests/lib/hardware_keyboard.dart [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        title: const Text('Hardware Key Demo'),
+      ),
+      body: const Center(
+        child: RawKeyboardDemo(),
+      ),
+    ),
+  ));
+}
+
+class RawKeyboardDemo extends StatefulWidget {
```
Comment (dkwingsmt) [inline dev/manual_tests/lib/hardware_keyboard.dart:22]: Rename to `HardwareKeyboardDemo`.

**Review by dkwingsmt — APPROVED** [review]
LGTM

File: dev/manual_tests/lib/hardware_keyboard.dart (hunk 1 above)
Comment (gspencergoog) [reply dev/manual_tests/lib/hardware_keyboard.dart:22]: Done.

**Pull request merged**